aten/src/ATen/native/native_functions.yaml

```diff
@@ -7098,31 +7098,57 @@
   use_c10_dispatcher: full
   variants: method, function
 
-- func: quantile.scalar_out(Tensor self, float q, int? dim=None, bool keepdim=False, *, Tensor(a!) out) -> Tensor(a!)
+# The following (nan)quantile signatures were deprecated in favor of the ones with interpolation parameter
```

Comment:
say why.

Reply:
there is a lot of subtlety going on here -- you are remapping the overload name to the "new' function and introducing these deprecated functions.  But are you sure this works for all the use cases?

E.g.:
- Calling with kwargs: if I do something like `quantile(self, q, dim=None, keepdim=False)` isn't that valid for both the deprecated and non-deprecated signatures?  At some point in eager we picked the earliest signature in the list, which would be the wrong thing in this case?  I also don't know what the JIT arg parsing would favor in this case.
- Serializing with TorchScript: do we serialize the overload name and the kwarg properties?
- Same questions with mobile serialization.

Reply:
`quantile(self, q, dim=None, keepdim=False)` does indeed call the deprecated signature.

Reply:
My thinking was that after some time of warning users of the deprecation we can remove the old signature without affecting users of the new signature. The only other option I know is to deprecate it now in which case it will be BC-breaking. Is there another option?

Reply:
In an offline discussion with @mruberry we decided to make the interpolation parameter kwarg only to prevent the BC breaking change.